.github/workflows/ci.yml

```diff
@@ -1,44 +1,64 @@
 name: Run CI
+
 on:
   push:
     branches:
       - master
   pull_request:
 
+env:
+  TZ: Asia/Singapore
+  NODE_OPTIONS: --max_old_space_size=8192
+  GENERATE_SOURCEMAP: false
+
 jobs:
-  deploy:
-    name: Run CI
+  deps:
+    name: Install dependencies
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup repository
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: yarn
+  lint:
+    needs: deps
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        commands: [tsc, 'format:ci', eslint, build]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Setup repository
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
```

Comment:
I agree but I think this is outside the scope of this PR. I recall seeing an issue/a mention somewhere about moving to Node 18 as Node 16 is approaching EOL this coming September.

I am planning to make a separate PR that:

* Creates the `.node-version` file
* Updates the README with the necessary instructions
* Updates the CI workflow above to use the `.node-version` file

This will also provide a concrete target to act on for the issue above.